src/openrct2/ride/Vehicle.cpp

```diff
@@ -6644,156 +6610,124 @@ void Vehicle::UpdateAnimationAnimalFlying()
 void Vehicle::UpdateAdditionalAnimation()
 {
     uint8_t targetFrame{};
-    uint8_t curFrame{};
-    uint32_t eax{};
-
     auto carEntry = Entry();
     if (carEntry == nullptr)
     {
         return;
     }
+    if (carEntry->AnimationFrames == 0)
+        return;
     switch (carEntry->animation)
```

Comment:
Could you perhaps move all those cases out into new functions? Having it all slammed into one function feels quite wrong.

Reply:
What is your preference for doing that? I can have function calls in a switch or I could put all the functions in an array and call based on `static_cast<uint8_t>(animation)` similar to choosing the vehicle paint pitch function.

Reply:
You can either keep the switch or use a function table with a common signature, both works for me. It will be a lot easier to make out the functionality of each animation type when they are clearly separated.